### PR TITLE
[14.0] account_payment_partner: refactor partner_banks_to_show

### PR DESCRIPTION
**Pull request by hparfr**
move partner_bank_to_show into account_payment_mode to allow it's reuse from other models like sale_order

depends of #1218

### CODE REVIEW

File: account_payment_partner/models/account_move.py [hunk 1]
```diff
@@ -142,22 +142,7 @@ def partner_banks_to_show(self):
         self.ensure_one()
         if self.partner_bank_id:
             return self.partner_bank_id
-        if self.payment_mode_id.show_bank_account_from_journal:
-            if self.payment_mode_id.bank_account_link == "fixed":
```
Comment (pedrobaeza) [inline account_payment_partner/models/account_move.py:146]: Include an override of `partner_banks_to_show` in SEPA DD module including this, indicating that the aim of this is to show the bank account where the SEPA DD is going to be charged.

Comment (hparfr) [reply account_payment_partner/models/account_move.py:146]: [account_banking_sepa_direct_debit](https://github.com/OCA/bank-payment/blob/14.0/account_banking_sepa_direct_debit/__manifest__.py#L14C47-L14C70) do not depend of this module.

Comment (pedrobaeza) [reply account_payment_partner/models/account_move.py:146]: Then that's the reason for doing such "ugly thing". We need to show the customer account for SEPA DD.

Comment (hparfr) [reply account_payment_partner/models/account_move.py:146]: The real thing would have to create a glue module for that.

Anyway, this small change (<10 len if), start to cost me a lot in refactorings for features I don't need.

Comment (pedrobaeza) [reply account_payment_partner/models/account_move.py:146]: Yeah, the feature was there before the refactoring. We can't lose it. I think we can keep the hack without doing a glue module, although it doesn't seem very elegant.

**Comment by hparfr** [comment]
glue module added

**Review by florian-dacosta — APPROVED** [review]
LGTM

**Pull request closed without merging**